certbot_route53/authenticator.py

```diff
@@ -53,6 +53,11 @@ def get_chall_pref(self, domain):
     def perform(self, achalls):  # pylint: disable=missing-docstring
         try:
             change_ids = [self._create_single(achall) for achall in achalls]
+            # Sleep for at least the TTL, to ensure that any records cached by
+            # the ACME server after previous validation attempts are gone. In
+            # most cases we'll need to wait at least this long for the Route53
+            # records to propagate, so this doesn't delay us much.
+            time.sleep(TTL)
```

Comment:
The sleep has to happen after all the calls to `_wait_for_change` to achieve the intended purpose. You first have to wait for route53 to propagate internally (that's what `_wait_for_change` does for us), then wait for outside resolver to expire the cached values.